nnvm/python/nnvm/top/nn.py

```diff
@@ -170,7 +170,10 @@ def compute_contrib_conv2d_NCHWc(attrs, inputs, _):
     out_layout = attrs.get_string("out_layout")
     out_dtype = attrs.get_string("out_dtype")
     out_dtype = inputs[0].dtype if out_dtype == "same" else out_dtype
-    _, in_channel_chunk, _, _, in_channel_block = get_const_tuple(inputs[0].shape)
+    if len(get_const_tuple(inputs[0].shape)) == 5:
+        _, in_channel_chunk, _, _, in_channel_block = get_const_tuple(inputs[0].shape)
+    else:
+        _, in_channel_chunk, _, _, in_channel_block = get_const_tuple(inputs[1].shape)
```

Comment:
suggest to test attr.layout instead of the size of shape, so that we can make it more clear and more general. Also in else stmt, it is better to get tuple from data instead of kernel, unless you check kernel layout here.